src/uct/ib/base/ib_md.c

```diff
@@ -452,6 +474,19 @@ ucs_status_t uct_ib_reg_mr(uct_ib_md_t *md, void *address, size_t length,
     dmabuf_offset = UCS_PARAM_VALUE(UCT_MD_MEM_REG_FIELD, params, dmabuf_offset,
                                     DMABUF_OFFSET, 0);
 
+#if HAVE_IBV_DM && HAVE_DECL_IBV_IMPORT_DM
+    if (dm_handle != UCT_IB_MD_NO_DM_HANDLE) {
+        struct ibv_dm *dm = ibv_import_dm(md->dev.ibv_context, dm_handle);
+        access_flags      = UCT_IB_MEM_ACCESS_FLAGS | IBV_ACCESS_ZERO_BASED;
+        title             = "ibv_reg_dm_mr";
+        mr = UCS_PROFILE_CALL_ALWAYS(ibv_reg_dm_mr, md->pd, dm, 0, length,
+                                     access_flags);
+        if (mr != NULL) {
+            mr->addr   = address;
+            mr->length = length;
```

Comment:
we should not override structs returned by verbs layer
recently @ivankochin added address to memh, we can use it